src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarily.cs

```diff
@@ -0,0 +1,104 @@
+                    {
+                        // Skip through built-in conversions
+                        IOperation value = init.Value;
+                        while (value is IConversionOperation conversion && !conversion.Conversion.IsUserDefined)
+                        {
+                            value = conversion.Operand;
+                        }
+
+                        // If this is default(T) or new ValueType(), it's the default.
+                        if (value is IDefaultValueOperation ||
+                            (field.Type.IsValueType && value is IObjectCreationOperation oco && oco.Arguments.Length == 0))
+                        {
+                            return !IsNullSuppressed(value);
+                        }
+
+                        // Then if this isn't a literal, it's not the default.
+                        if (!(value is ILiteralOperation literal) || !literal.ConstantValue.HasValue)
+                        {
+                            return false;
+                        }
+
+                        // If this is a reference type or a nullable, it's the default if the value is null.
+                        if (field.Type.IsReferenceTypeOrNullableValueType())
+                        {
+                            return literal.ConstantValue.Value is null && !IsNullSuppressed(literal);
+                        }
+
+                        // Finally, if this is a primitive (including enums), it's the default if it's 0/false.
+                        return
+                            (field.Type.IsPrimitiveType() || field.Type.TypeKind == TypeKind.Enum) &&
+                            literal.ConstantValue.Value switch
+                            {
+                                bool b => b == false,
+                                char c => c == 0,
+                                byte b => b == 0,
+                                sbyte s => s == 0,
+                                ushort u => u == 0,
+                                short s => s == 0,
+                                uint u => u == 0,
+                                int i => i == 0,
+                                ulong u => u == 0,
+                                long l => l == 0,
+                                float f => f == 0f,
+                                double d => d == 0,
+                                _ => false
+                            };
+
+                        // Special-case `null!`/`default!` to not warn about it, as it's often used to suppress nullable warnings on fields.
+                        static bool IsNullSuppressed(IOperation op) =>
+                            op.Syntax?.Parent is PostfixUnaryExpressionSyntax parent &&
+                            parent.Kind() == CodeAnalysis.CSharp.SyntaxKind.SuppressNullableWarningExpression;
+                    }
+                }
+            },
+            OperationKind.FieldInitializer);
```

Comment:
Should this rule also apply to property and local symbol initializers?

Reply:
Actually not local symbol initializers due to definite assignment requirements, but seems reasonable to include property initializers?

Reply:
Makes sense.  Added.